src/mame/barcrest/mpu4_oki_sampled_sound.h

```diff
@@ -0,0 +1,66 @@
+// license:BSD-3-Clause
+// copyright-holders:David Haywood, James Wallace
+
+#ifndef MAME_MACHINE_MPU4_OKI_SAMPLED_SOUND_H
+#define MAME_MACHINE_MPU4_OKI_SAMPLED_SOUND_H
+
+#pragma once
+
+#include "machine/6821pia.h"
+#include "machine/6840ptm.h"
+#include "sound/okim6376.h"
+
+
+#ifdef MAME_DEBUG
+#define MPU4SSVERBOSE 1
+#else
+#define MPU4SSVERBOSE 0
+#endif
+
+#define LOG_SS(x)   do { if (MPU4SSVERBOSE) logerror x; } while (0)
+
+DECLARE_DEVICE_TYPE(MPU4_OKI_SAMPLED_SOUND, mpu4_oki_sampled_sound)
+
+class mpu4_oki_sampled_sound : public device_t, public device_mixer_interface
+{
+public:
+	// construction/destruction
+	mpu4_oki_sampled_sound(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock = 0);
+
+	void ic3_write(offs_t offset, uint8_t data);
+	uint8_t ic3_read(offs_t offset);
+
+	void ic4_write(offs_t offset, uint8_t data);
+	uint8_t ic4_read(offs_t offset);
+
+	auto cb2_handler() { return m_cb2_handler.bind(); }
```

Comment:
Please keep configuration member functions like this one above the live signal functions.